mmcv/runner/dist_utils.py

```diff
@@ -64,8 +84,12 @@ def _init_dist_slurm(backend, port=None):
     elif 'MASTER_PORT' in os.environ:
         pass  # use MASTER_PORT in the environment variable
     else:
-        # 29500 is torch.distributed default port
-        os.environ['MASTER_PORT'] = '29500'
+        # if torch.distributed default port(29500) is available
+        # then use it, else find a free port
+        if _is_free_port(29500):
+            os.environ['MASTER_PORT'] = '29500'
+        else:
+            os.environ['MASTER_PORT'] = str(_find_free_port())
```

Comment:
1. we should log  the updated free port to inform users since it is different from the expected port
2. what if the port set by users is not free? This PR does not fix such a case, which is also a common case.